src/frontend/mosh-server.cc

```diff
@@ -336,6 +339,26 @@ int main( int argc, char *argv[] )
 static int run_server( const char *desired_ip, const char *desired_port,
 		       const string &command_path, char *command_argv[],
 		       const int colors, bool verbose, bool with_motd ) {
+  /* get network idle timeout */
+  long network_timeout = 0;
+  char *timeout_envar = getenv( "MOSH_SERVER_NETWORK_TMOUT" );
+  if ( timeout_envar ) {
+    /* We don't care about strtol errors here, they all produce sensible return values. */
+    network_timeout = strtol( timeout_envar, NULL, 10 );
+    if ( network_timeout < 0 ) {
+      network_timeout = 0;
```

Comment:
If there is an error, this value gets set to 0, but in line 718 below 0 is treated as "stay alive indefinitely". This might bite users - they intended something else, but aren't told about the error. Granted, it looks like this is exactly what bash does with `TMOUT` but it's not a good principle to follow. One solution is to have 0 actually mean "exit immediately". If you don't want to do that, you should at least make the docs more precise:

- When set to an integer number -> When set to a positive integer number
- When not set -> when not set, or set to an invalid or non-positive value